apps/builder/app/routes/tmp.login.tsx

```diff
@@ -0,0 +1,244 @@
+import { json, type LoaderFunctionArgs } from "@remix-run/server-runtime";
+import { authenticator } from "~/services/auth.server";
+import { builderAuthenticator } from "~/services/builder-auth.server";
+import { returnToCookie } from "~/services/cookie.server";
+import { createContext } from "~/shared/context.server";
+import {
+  getAuthorizationServerOrigin,
+  isBuilderUrl,
+  parseBuilderUrl,
+} from "~/shared/router-utils/origins";
+import { Root } from "~/shared/remix/root";
+import createDebug from "debug";
+import { builderUrlNew } from "~/shared/router-utils/path-utils";
+
+const debug = createDebug("Auth").extend("TmpLogin");
+
+export const meta: MetaFunction = () => {
+  return [
+    { title: "Temp login page (to delete)" },
+    { name: "description", content: "Temp login page!" },
+  ];
+};
+
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  debug("request.url", request.url);
+  debug("isBuildUrl", isBuilderUrl(request.url));
+  debug("Headers", [...request.headers.entries()]);
+  debug("Origin", getAuthorizationServerOrigin(request.url));
+
+  const context = await createContext(request);
+
+  const projectData = await context.postgrest.client
+    .from("Project")
+    .select("id")
+    .order("createdAt", { ascending: false })
+    .limit(1)
+    .single();
+  if (projectData.error) {
+    throw new Error(
+      `Failed to fetch project data: ${projectData.error.message}`
+    );
+  }
+
+  const projectId = isBuilderUrl(request.url)
+    ? parseBuilderUrl(request.url).projectId!
+    : process.env.NODE_ENV === "production"
+      ? "264a609f-ce63-4197-a364-68d239c2f56a"
+      : projectData.data.id;
+
+  const user = isBuilderUrl(request.url)
+    ? await builderAuthenticator.isAuthenticated(request)
+    : await authenticator.isAuthenticated(request);
+
+  const headers = new Headers();
+
+  if (isBuilderUrl(request.url)) {
+    debug("Temporary redirect to /tmp/login after login");
+    const returnTo = request.url;
+    headers.append("Set-Cookie", await returnToCookie.serialize(returnTo));
+  }
+
+  return json(
+    {
+      user,
+      projectId,
+      isBuilderUrl: isBuilderUrl(request.url),
+      origin: getAuthorizationServerOrigin(request.url),
+    },
+    { headers }
+  );
+};
+
+const TmpLogin = () => {
+  const data = useLoaderData<typeof loader>();
+
+  const projectUrl = new URL(
+    builderUrlNew({
+      origin: data.origin,
+      projectId: data.projectId,
+    })
+  );
+
+  projectUrl.pathname = "/tmp/login";
+
+  const noAccessProjectUrl = new URL(
+    builderUrlNew({
+      origin: data.origin,
+      projectId: "9ccd9600-de48-4f3f-898b-042e890ae805",
+    })
+  );
+
+  noAccessProjectUrl.pathname = "/tmp/login";
+
+  const fetchAllProjectIdsLoggedInAfterIssueDate = async (
+    _userId: string,
+    _issueDate: number
+  ) => {
+    return [
+      {
+        id: data.projectId,
+      },
+    ];
+  };
+
+  return (
+    <div className="font-sans p-4">
```

Comment:
how did you get tailwind in here :)

Reply:
see file name, its created to test current implementation